### DISCUSSION TITLE
Google Code Scanner API - Initial orientation

### DISCUSSION THREAD

**Discussion by reisier** [googlesamples/mlkit#865] [General]
In an application that works in landscape mode, the scanner starts in portrait mode and then switches to landscape and after the qr code is read it switches back to landscape.
Is there any way to tell the initial orientation of the scanner and fix this orientation?
Thank you

**Comment by sweetdream-001** [comment]
1. Lock your app's orientation to landscape mode before starting the scanner.

2. Use Android's setRequestedOrientation() method to force landscape:

```
java
setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
```
3. Start the scanner after setting the orientation.

4. When scanning is complete, you can unlock the orientation if needed.

This should keep your app and the scanner in landscape mode throughout the process, avoiding unwanted orientation changes.
